students/shirin_ak/Session09/donor_models.py

```diff
@@ -0,0 +1,142 @@
+import sys
+import string
+
+#This class will hold all the information about a single donor,
+#and have attributes, properties, and methods to provide access
+#to the donor-specific information.
+
+class Donor():
+    """Donor Class"""
+    def __init__(self, name):
+        self.name = name
+        self.donation = []
+
+    def __name__(self):
+        return self.name
+    
+    def add_donations(self, amount):
+        self.donation.append(amount)
+
+        
+    @property
+    def total_donations(self):
+        """return total donation of a single donor"""
+        return sum(self.donation)
+
+    
+    @property
+    def num_donations(self):
+        """return number of donations a donor donated"""        
+        return len(self.donation)
+
+    
+    @property
+    def ave_donations(self):
+        return self.total_donations/self.num_donations
+
+    
+    def __str__(self):
+        return f'{self.name}:{self.donation}'
+
+    
+    @property
+    def thank_you_letter(self):
+        """
+        returns a string that is a formatted thank you note
+        with the donor's name and last donation amount.
+        """
+        return "Dear {:s},\n\nWe greatly appreciate your generous donation of"
+        " ${:,.2f}.\n\nThank you,\nThe Team".format(self.name, self.donations[-1])
+        
+ #--------------------------------
+ #       End of Donor class
+ #------------------------------
+
+# This class will hold all of the donor objects, as well as methods to add a new donor,
+#search for a given donor,  generates reports about multiple donors.
+#In other words,if the functionality involves more than one donor – it belongs in this class.
+    
+class DonorCollection():
+    def __init__(self):
+        self.donors_dict = {}
+                            
+                       
+    def find_donor(self, name):
+        if name.lower() in self.donors_dict:
+            return self.donors_dict[name.lower()]
+        return None
+
+
+    def add_donor(self, name):
+        """allows for adding new donors to the db"""
+        
+        new_donor = Donor(name)
+        self.donors_dict[new_donor.name.lower()] = new_donor
+        return new_donor
+
+    
+    @staticmethod
+    def sort_key(item):
+        return item[1]
+
+    
+    def donor_list(self):
+        name_list = [donor.name for donor in self.donors_dict]
+        return '\n'.join(name_list)
+
+
+#Builds the donor report and displays the name, total given, number of gifts
+# and the average gift donated by the donor.
+
+    def donor_report(self):
+        report_rows = []
+        for donor in self.donors_dict:
```

Comment:
You want _donor_ here to be the an object of class _Donor_. The way it's currently written, it will return the key, a _str_ object. Try instead:

for donor in self.donors_dict.values()